webtau-graphql/src/main/java/org/testingisdocumenting/webtau/graphql/listener/GraphQLListeners.java

```diff
@@ -0,0 +1,87 @@
+package org.testingisdocumenting.webtau.graphql.listener;
+
+import org.testingisdocumenting.webtau.graphql.model.GraphQLRequest;
+import org.testingisdocumenting.webtau.graphql.model.GraphQLResponse;
+import org.testingisdocumenting.webtau.http.HttpHeader;
+import org.testingisdocumenting.webtau.http.HttpResponse;
+import org.testingisdocumenting.webtau.http.listener.HttpListener;
+import org.testingisdocumenting.webtau.http.listener.HttpListeners;
+import org.testingisdocumenting.webtau.http.request.HttpRequestBody;
+import org.testingisdocumenting.webtau.utils.ServiceLoaderUtils;
+import org.testingisdocumenting.webtau.utils.UrlUtils;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class GraphQLListeners {
+    private static final List<WrappedGraphQLListener> listeners = ServiceLoaderUtils.load(GraphQLListener.class)
+            .stream().map(WrappedGraphQLListener::new).collect(Collectors.toList());
+
+    private GraphQLListeners() {
+    }
+
+    public static void beforeFirstGraphQLQuery() {
+        listeners.forEach(l -> l.listener.beforeFirstGraphQLQuery());
+    }
+
+    public static void beforeGraphQLQuery(String query,
+                                          Map<String, Object> variables,
+                                          String operationName,
+                                          HttpHeader requestHeader) {
+        listeners.forEach(l -> l.listener.beforeGraphQLQuery(query, variables, operationName, requestHeader));
+    }
+
+    public static void add(GraphQLListener listener) {
+        listeners.add(new WrappedGraphQLListener(listener));
+    }
+
+    public static void remove(GraphQLListener listener) {
+        listeners.stream()
+                .filter(l -> l.listener == listener)
+                .findFirst()
+                .ifPresent(l -> {
+                    HttpListeners.remove(l);
+                    listeners.remove(l);
+                });
+    }
+
+    private static class WrappedGraphQLListener implements GraphQLListener, HttpListener {
```

Comment:
I chose to do this wrapping approaching because it was the most elegant way to actually get access to the response.  I could have called the after callback from within `GraphQL.execute` but I only have a `DataNode` in there.